### PR TITLE
wip: add getAtomQueryFeaturesObject

### PR DESCRIPTION
**Pull request by lpatiny**

### CODE REVIEW

File: __tests__/molecule.queryFeatrues.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+
+const { Molecule } = require('../minimal')
+
+
+test('queryFeatures', () => {
+  const molecule = Molecule.fromSmiles('CCO');
+  const features = molecule.getAtomQueryFeaturesObject(0);
+  console.log(features)
```
Comment (lpatiny) [inline __tests__/molecule.queryFeatrues.test.js:10]: Currently the return value is an empty object `{}`

File: src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.actelion.research.gwt.minimal;
+
+import jsinterop.annotations.*;
+import com.actelion.research.chem.StereoMolecule;
+
```
Comment (lpatiny) [inline src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java:5]: I would like to easily create a JavaScriptObject from a function but don't know how to do.

File: src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+class MoleculeQueryFeatures {
+  String oclID;
+  String smiles;
+
+  @JsOverlay
+  public static MoleculeQueryFeatures getAtomQueryFeatures(StereoMolecule oclMolecule, int atom) {
+    MoleculeQueryFeatures moleculeQueryFeatures = new MoleculeQueryFeatures();
+    moleculeQueryFeatures.oclID = "ABC";
+    // moleculeQueryFeatures.oclID = Long.toString(oclMolecule.getAtomQueryFeatures(0));
+    moleculeQueryFeatures.smiles = "asd";
```
Comment (lpatiny) [inline src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java:16]: Could contain string, boolean or numbers

**Review by lpatiny — COMMENTED** [review]
@targos I could not find a good way to return a `JavaScriptObject` from a Java class. I wanted to do this to expose the queryFeatures of the atoms. Could you help me ?

**Comment by targos** [comment]
It depends what you want to return. If you want a plain JS object, you should not create a new Java class.

File: src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import com.actelion.research.chem.StereoMolecule;
+import com.actelion.research.chem.Molecule;
+import com.google.gwt.core.client.JavaScriptObject;
+
+class MoleculeQueryFeatures {
+  public static JavaScriptObject getAtomQueryFeatures(StereoMolecule oclMolecule, int atom) {
+    PlainJSObject moleculeQueryFeatures = PlainJSObject.create();
+
+    moleculeQueryFeatures.setPropertyBoolean("aromatic",
+        (oclMolecule.getAtomQueryFeatures(atom) & Molecule.cAtomQFAromatic) > 0);
```
Comment (targos) [inline src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java:12]: Wouldn't it be significantly faster to call `getAtomQueryFeatures(atom)` only once and put the value in a local variable?

Comment (lpatiny) [reply src/com/actelion/research/gwt/minimal/MoleculeQueryFeatures.java:12]: Indeed good idea! The method `getAtomQueryFeatures` is super small so probably small impact but the code will look any much better and we will have the possibility to optimise the part that is 'int' and not long. Thanks.
```
public long getAtomQueryFeatures(int atom) {
return mAtomQueryFeatures[atom];
}
```

**Pull request merged**